core/src/main/java/org/elasticsearch/cluster/ClusterStateObserver.java

```diff
@@ -307,13 +309,25 @@ public ObservingContext(Listener listener, ChangePredicate changePredicate) {
         }
     }
 
-    static class ObservedState {
-        public final ClusterState clusterState;
-        public final ClusterState.ClusterStateStatus status;
+    public static class ObservedState {
```

Comment:
For now, the `ObservedState` contains the same info as `ClusterServiceState`.  I also think anything we add in the future will likely also be relevant for the `ClusterStateObserver`, so I'm wondering if we need to maintain the `ObservedState` class and can just use `ClusterServiceState`.  I don't believe its a big issue though, so if you prefer to keep it as is, we can do that and revisit later if needed.